httpcore/_async/connection.py

```diff
@@ -99,3 +113,4 @@ def mark_as_ready(self) -> None:
     async def start_tls(self, hostname: bytes, timeout: TimeoutDict = None) -> None:
         if self.connection is not None:
             await self.connection.start_tls(hostname, timeout)
+            self.socket = self.connection.socket
```

Comment:
Would it make sense to modify `.start_tls()` so that it returns the `socket`, replicating the signature of `asyncio.start_tls()`? So that we're then able to write…

```suggestion
            self.socket = await self.connection.start_tls(hostname, timeout)
```

Reply:
I agree, but frankly, I think the managing of the socket needs to be looked at at a design level. Keeping references of it at three different abstraction levels is a strong smell.

That being said I'd like to land this fix for tunnel proxies so we have a solid testing suite to allow us to refactor.

Reply:
Sounds good :+1: